lib/app/features/feed/views/components/post/components/post_body/post_body.dart

```diff
@@ -62,31 +64,44 @@ class PostBody extends HookConsumerWidget {
       key: ValueKey(postData.hashCode),
     );
 
-    final firstLinkOperation = useMemoized(
-      () => content.operations.firstWhereOrNull(
-        (operation) => isAttributedOperation(operation, attribute: Attribute.link),
-      ),
+    final firstUrlInPost = useMemoized(
+      () {
+        final firstOperationLink = content.operations
+            .firstWhereOrNull(
+              (operation) => isAttributedOperation(operation, attribute: Attribute.link),
+            )
+            ?.value as String?;
+
+        return firstOperationLink == null ? null : normalizeUrl(firstOperationLink);
+      },
       [content],
     );
 
-    final urlPreviewVisible = useState(false);
-
-    final urlPreview = useMemoized(
-      () => firstLinkOperation != null
-          ? UrlPreviewContent(
-              key: ValueKey(firstLinkOperation.value),
-              url: firstLinkOperation.value as String,
-              onPreviewVisibilityChanged: (previewVisible) {
-                urlPreviewVisible.value = previewVisible;
-              },
-            )
-          : null,
-      [firstLinkOperation],
+    final hasValidUrlMetadata = useMemoized(
+      () =>
+          firstUrlInPost != null &&
+          (ref.watch(urlMetadataProvider(firstUrlInPost)).valueOrNull?.title?.isNotEmpty ?? false),
+      [firstUrlInPost],
     );
 
     final showTextContent = useMemoized(
-      () => content.isNotEmpty && (!content.isSingleLinkOnly || !urlPreviewVisible.value),
-      [content, urlPreviewVisible.value],
+      () {
+        if (content.isEmpty) {
+          return false;
+        }
+
+        if (content.isSingleLinkOnly) {
+          if (hasValidUrlMetadata) {
+            if (media.isEmpty) {
+              return false;
+            }
+            return true;
+          }
+        }
+
+        return true;
```

Comment:
Could it be simplified to this?

```suggestion
        if (content.isSingleLinkOnly && hasValidUrlMetadata && media.isEmpty) {
           return false;
        }
        return true;
```

Reply:
you are right, i just fixed